mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/HighLevelMqttSource.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.impl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.alpakka.mqtt.streaming.scaladsl._
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, RestartFlow, Source, SourceQueueWithComplete}
+import akka.stream.{Materializer, OverflowStrategy}
+import akka.{Done, NotUsed}
+
+import scala.collection.immutable
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Internal API
+ */
+@InternalApi
+private[streaming] object HighLevelMqttSource {
+
+  def atMostOnce(
+      mqttClientSession: MqttClientSession,
+      transportSettings: MqttTransportSettings,
+      restartSettings: MqttRestartSettings,
+      connectionSettings: MqttConnectionSettings,
+      subscriptions: MqttSubscribe
+  ): Source[Publish, Future[immutable.Seq[(String, ControlPacketFlags)]]] = {
+    type Out = Publish
+
+    val sendAcknowledge: SourceQueueWithComplete[Command[Nothing]] => PartialFunction[Event[Nothing], Out] =
+      commands => {
+        case Event(publish @ Publish(_, _, Some(packetId), _), _) =>
+          commands.offer(Command(PubAck(packetId)))
```

Comment:
I don't think you want to `PubAck` here - see [this table](https://docs.oasis-open.org/mqtt/mqtt/v5.0/cos02/mqtt-v5.0-cos02.html#_Toc1477420) for expected replies (none in QoS0)

Reply:
It might not be intended use, but in a test "At-most-once - should receive subscribed messages only once (AtLeastOnceDelivery)", I use a QoS1 subscription on this `atMostOnce` source. When I don't do this acking it doesn't progress after the first element...

Reply:
I may not be understanding, but QoS 0 isn’t guaranteed to receive anything at all...